### PR TITLE
Slack notifications with slackrus

### PR DESCRIPTION
**Pull request by ubergesundheit**
This pull request implements and #61 with slackrus.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -114,6 +114,17 @@ func main() {
...
 			EnvVar: "WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PASSWORD",
 		},
+		cli.StringFlag{
+			Name:   "notification-slack-hook-url",
+			Usage:  "The Slack Hook URL to send notifications to",
+			EnvVar: "WATCHTOWER_NOTIFICATION_SLACK_HOOK_URL",
+		},
+		cli.StringFlag{
+			Name:   "notification-slack-identifier",
+			Usage:  "A string which will be used to identify the messages comming from this watchtower instance. Default if omitted is \"watchtower\"",
```
Comment (datashaman) [inline main.go]: comming -> coming

File: glide.lock [hunk 2]
```diff
@@ -135,10 +138,11 @@ imports:
 - name: github.com/robfig/cron
   version: 9585fd555638e77bba25f25db5c44b41f264aeb7
 - name: github.com/Sirupsen/logrus
```
Comment (stffabi) [inline glide.lock:141]: We should remove this old logrus entry with the uppercase "Sirupsen".

File: notifications/slack.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package notifications
+
+import (
+	"github.com/johntdyer/slackrus"
+	"github.com/sirupsen/logrus"
```
Comment (stffabi) [inline notifications/slack.go]: logrus should only be imported once.

**Review by stffabi — CHANGES_REQUESTED** [review]
Thanks so much for your contribution. I've two minor comments, apart from that LGTM.

**Comment by stffabi** [comment]
One more thing, could you also add some documentation? That would be great.

File: glide.lock (hunk 2 above)
Comment (ubergesundheit) [reply glide.lock:141]: It seems at least docker and go-ansiterm are using the uppercase version of "Sirupsen".
Removing it manually the uppercase entry is re-added after running `glide update`

**Comment by ubergesundheit** [comment]
So what should we do about the duplicate entry of `Sirupsen`/`sirupsen`?

File: notifications/slack.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+type slackTypeNotifier struct {
+	slackrus.SlackrusHook
+}
+
+func newSlackNotifier(c *cli.Context) typeNotifier {
+	n := &slackTypeNotifier{
+		SlackrusHook: slackrus.SlackrusHook{
+			HookURL:        c.GlobalString("notification-slack-hook-url"),
+			Username:       c.GlobalString("notification-slack-identifier"),
+			AcceptedLevels: slackrus.LevelThreshold(log.DebugLevel),
```
Comment (rdamazio) [inline notifications/slack.go]: Note that with DebugLevel you'll potentially get a lot of output - e.g. every time it checks for updates, even if there are none. Should it at least be configurable?

Comment (stffabi) [reply notifications/slack.go]: @rdamazio do you know if the AcceptedLevels are independent of [setLogLevel](https://github.com/v2tec/watchtower/blob/master/main.go#L131)? So basically even if we have a non Debug-loglevel, will a Hook with threshold `DebugLevel` be triggered with Debug notifications?

**Comment by Nightbr** [comment]
hey, thanks for you're work! Any news on this, really need the slack notification for watchtower a must have I think.

For the uppercase of sirupsen, if you look at logrus repo, they use sirupsen without uppercase in latest version. So I thin we should go with `sirupsen` in watchtower code.

**Comment by stffabi** [comment]
@Nightbr I hope to find some time this week to review the open PRs for `v0.3.0`.

**Review by stffabi — APPROVED** [review]
LGTM, I've squashed some of your commits and made the nofitication level a global setting for all notifications types.

Thanks for your great contribution.

**Comment by ubergesundheit** [comment]
Very nice! Thanks

**Comment by Nightbr** [comment]
Thanks!

**Pull request merged**